### PR TITLE
Demonstrate using globals in templating

### PR DESCRIPTION
**Pull request by michaeldfallen**
This PR demonstrates how globals and multiple extends can be used to clean up templates.

**Globals**

Any variables defined in `nunjucks.globals` are available in all templates. This is useful for things that appear on all templates like the phase, service name or the feedback link.

```
lookAndFeel.configure(app, {
  ...
  nunjucks: {
    globals: {
      phase: 'ALPHA',
      feedbackLink: 'https://github.com/hmcts/one-per-page/issues/new'
    }
  }
  ...
});
```

**Multiple extends**

You can extend more than one template in your templates. The last extend will provide the document root. 

This means that you need to put the layout you want to extend last and any extends before this can only set variables and implement blocks. 

The blocks in those earlier extends will override the default blocks set in the layout. You can still of course redeclare these blocks in your template itself.

In the example app we use this to set some blocks that we extend in all templates, the service name in the header and linking the css and js:

*question-globals.njk*:
```
{% block proposition_header %}
  {% from "look-and-feel/components/nav.njk" import propositionalNavigation %}
  {{ propositionalNavigation('Apply for Divorce') }}
{% endblock %}

{% block header_class %}
  with-proposition
{% endblock %}

{% block head -%}
<link href="{{ asset_path }}main.css" media="screen" rel="stylesheet" />
<script type="text/javascript" src="{{ asset_path }}main.js"></script>
{% endblock %}
```

*name.template.html*
```
extends "question-globals.njk"
extends "look-and-feel/layouts/question.html"

{% block fields %}
  ...
{% endblock %}
```

### CODE REVIEW

File: examples/test-app/app.js [hunk 1]
```diff
@@ -20,12 +20,23 @@ const baseUrl = `http://localhost:${config.port}`;
 
 lookAndFeel.configure(app, {
   baseUrl,
-  express: { views: [path.resolve(__dirname, 'steps')] },
+  express: {
+    views: [
+      path.resolve(__dirname, 'steps'),
+      path.resolve(__dirname, 'views')
```
Comment (endamccormack) [inline examples/test-app/app.js]: is this related to this pull request?

File: examples/test-app/steps/Country.content.json [hunk 2]
```diff
@@ -10,8 +10,7 @@
         "northernIreland": "Northern Ireland",
```
Comment (endamccormack) [inline examples/test-app/steps/Country.content.json]: norn'Ireland*

File: examples/test-app/views/question-globals.njk [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  {{ propositionalNavigation('Apply for Divorce') }}
+{% endblock %}
+
+{% block header_class %}
+  with-proposition
+{% endblock %}
+
+{% block head -%}
+<link href="{{ asset_path }}main.css" media="screen" rel="stylesheet" />
+<script type="text/javascript" src="{{ asset_path }}main.js"></script>
```
Comment (endamccormack) [inline examples/test-app/views/question-globals.njk]: Is it correct in assuming the asset path will contain the ending `/`

File: examples/test-app/app.js (hunk 1 above)
Comment (michaeldfallen) [reply examples/test-app/app.js]: Need to add that folder otherwise you can't do `{% extend "template.html" %}. Nunjucks is using expresses view folders as the files that are in scope for extends / imports / etc

File: examples/test-app/steps/Country.content.json (hunk 2 above)
Comment (michaeldfallen) [reply examples/test-app/steps/Country.content.json]: *nornIron*

File: examples/test-app/views/question-globals.njk (hunk 3 above)
Comment (michaeldfallen) [reply examples/test-app/views/question-globals.njk]: yeah, I kinda regret that.

Tempted to make a function in look-and-feel like rails has: `{{ asset('main.css') }}` which resolves the filepath for you. and then people will stop asking this question :wink: .
